R/derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,528 @@
+#'
+#'       - `"CR"`: An assessment is considered as complete response (CR) if
+#'           - `AVALC == "CR"`,
+#'           - there is a confirmatory assessment with `AVALC == "CR"` at least
+#'           `ref_confirm` days after the assessment,
+#'           - all assessments between the assessment and the confirmatory
+#'           assessment are `"CR"` or `"NE"`, and
+#'           - there are at most `max_nr_ne` `"NE"` assessments between the
+#'           assessment and the confirmatory assessment.
+#'
+#'       - `"PR"`: An assessment is considered as partial response (PR) if
+#'           - `AVALC == "PR"`,
+#'           - there is a confirmatory assessment with `AVALC %in% c("CR",
+#'           "PR")` at least `ref_confirm` days after the assessment,
+#'           - all assessments between the assessment and the confirmatory
+#'           assessment are `"CR"`, `"PR"`, or `"NE"`,
+#'           - there is no `"PR"` assessment after a `"CR"` assessment in the
+#'           confirmation period, and
+#'           - there are at most `max_nr_ne` `"NE"` assessments between the
+#'           assessment and the confirmatory assessment.
+#'
+#'           If the `accept_sd` argument is set to `TRUE`, one `"SD"` assessment
+#'           in the confirmation period is accepted.
+#'
+#'        - `"SD"`: An assessment is considered as stable disease (SD) if
+#'          - `AVALC %in% c("CR", "PR", "SD")` and
+#'          - the assessment is at least `ref_start_window` days after
+#'          `reference_date`.
+#'
+#'        - `"NON-CR/NON-PD"`: An assessment is considered as NON-CR/NON-PD if
+#'          - `AVALC = "NON-CR/NON-PD"` and
+#'          - the assessment is at least `ref_start_window` days after
+#'          `reference_date`.
+#'
+#'        - `"PD"`: An assessment is considered as progressive disease (PD) if
+#'        `AVALC == "PD"`.
+#'
+#'        - `"NE"`: An assessment is considered as not estimable (NE) if
+#'            - `AVALC == "NE"` or
+#'            - `AVALC %in% c("CR", "PR", "SD")` and the assessment is less than
+#'            `ref_start_window` days after `reference_date`.
+#'
+#'        - `"ND"`: An assessment is considered as not done (ND) if `AVALC ==
+#'        "ND"`.
+#'
+#'        - `"MISSING"`: An assessment is considered as missing (MISSING) if a
+#'        subject has no observation in the input dataset.
+#'
+#'            If the `missing_as_ne` argument is set to `TRUE`, `AVALC` is set to
+#'            `"NE"` for missing assessments.
+#'
+#'   1. For each subject the best response as derived in the previous step is
+#'   selected, where `"CR"` is best and `"MISSING"` is worst in the order above.
+#'   If the best response is not unique, the first one (with respect to `ADT`)
+#'   is selected. Only the variables `AVALC`, `ADT`, and the variables specified
+#'   for `subject_keys` are kept from the selected observations.
```

Comment:
i would keep all the ADSL vars as per our discussion with Andrew on the other PR, i can't see any case where i would want these to be left blank for newly created records

Reply:
Updated as agreed on in https://github.com/pharmaverse/admiralonco/issues/24.